File: 프로그래머스/2/42587. 프로세스/프로세스.js
function max(array){
    var ret = 0;
    for(var elem of array){
        if(ret < elem[1]){
            ret = elem[1];
        }
    }
    return ret;
}

function solution(priorities, location) {
    var answer = 0;
    var priorityIndex = []
    
    for(var i = 0; i < priorities.length ; i ++){
        priorityIndex.push([i, priorities[i]]);
    }
    
    var maxValue = max(priorityIndex)

    while (priorityIndex.length > 0){
        var elem = priorityIndex.shift();
        if(elem[1] === maxValue){
            maxValue = max(priorityIndex)
            answer += 1
            if (elem[0]===location){
                return answer
            }
        }
        else{
            priorityIndex.push(elem)
        } 
    }    
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    test1 = []
    test2 = []
    for i in priorities:
        test1.append(i)
        test2.append(i)
    test2[location] = -1
    answer = 1
    if test2[0] == -1 and max(test1) == test1[0]:
        return answer

    while test1:
        print(test1, test2, answer)
        if test2[0] == -1 and max(test1) == test1[0]:
            return answer

        elif max(test1) == test1[0]:
            answer = answer + 1
            test1.pop(0)
            test2.pop(0)
        else:
            test1.append(test1.pop(0))
            test2.append(test2.pop(0))

    return answer

print(solution([1, 1, 9, 1, 1, 1], 0))